apps/go-mod-validator/src/deps.ts

```diff
@@ -0,0 +1,47 @@
+import { execSync } from "child_process";
+import * as core from "@actions/core";
+
+function getGoModFiles(goModDir: string): string[] {
+  let output = "";
+
+  try {
+    output = execSync(`find ${goModDir} -type f -name 'go.mod'`, {
+      encoding: "utf-8",
+    });
+  } catch (error) {
+    throw new Error(`failed to get go.mod files: ${error}`);
+  }
+
+  if (output.length == 0) {
+    throw new Error("no go.mod files found");
+  }
+  return output.trim().split("\n");
+}
+
+export function getDependenciesMap(goModDir: string): Map<string, any> {
+  // get all go.mod files
+  const modFilePaths = getGoModFiles(goModDir);
+
+  const dependenciesMap = new Map();
+  modFilePaths.forEach((modFilePath: string) => {
+    core.info(`finding dependencies in ${modFilePath}`);
+
+    // get the dir of go.mod file
+    const modFileDir = modFilePath.slice(0, -6);
```

Comment:
it'd be more useful to instead have the comment include an example `modFilePath` so that readers know what you're slicing. im not sure why you're slicing the last ~6 chars

Reply:
Done